rust_src/src/frames.rs

```diff
@@ -155,4 +156,27 @@ pub fn framep(object: LispObject) -> LispObject {
     }
 }
 
+/// Return t if FRAME is \"visible\" (actually in use for display).
+/// Return the symbol `icon' if FRAME is iconified or \"minimized\".
+/// Return nil if FRAME was made invisible, via `make-frame-invisible'.
+/// On graphical displays, invisible frames are not updated and are
+/// usually not displayed at all, even in a window system's \"taskbar\".
+///
+/// If FRAME is a text terminal frame, this always returns t.
+/// Such frames are always considered visible, whether or not they are
+/// currently being displayed on the terminal.
+#[lisp_fn]
+pub fn frame_visible_p(frame: LispObject) -> LispObject {
+    let frame_ref = frame.as_live_frame_or_error();
```

Comment:
It is common elsewhere in the tree to just mask the incoming variable. If you add methods as I suggest this will be possible.

Reply:
Forgot about that one.

Reply:
I'm not sure I understand correctly. You mean shadow `frame` ?

Reply:
Yes

Reply:
Hmmm. Is there really value in changing 
`let frame_ref = frame.as_live_frame_or_error();` to 
`frame = frame.as_live_frame_or_error();`

Also the type of `frame` is changed and I think the additional variable indicates it. And this is the way we do it in all the other functions.